File: pyforms/terminal/Controls/ControlText.py
from pyforms.terminal.Controls.ControlBase import ControlBase

class ControlText(ControlBase):

    def initControl(self):
        #return """<label for="id%s">%s</label><input type='text' name='%s' id='id%s' />""" % ( self._name,  self._label, self._name, self._name, )
        return "controls.push(new ControlText('%s','%s'));" % (self._label, self._name)
     

    ############################################################################
    ############ Properties ####################################################
    ############################################################################

    @property
    def value(self): return self._value

    @value.setter
    def value(self, value):
        ControlBase.value.fset(self,value)
        
        
    

File: pyforms/web/django/static/pyformsjs/ControlText.js


function ControlText(name, properties){
	ControlBase.call(this, name, properties);
};
ControlText.prototype = Object.create(ControlBase.prototype);


////////////////////////////////////////////////////////////////////////////////

ControlText.prototype.init_control = function(){
	var html = "<div id='"+this.place_id()+"' class='field ControlText' ><label>"+this.properties.label+"</label><input placeholder='"+this.properties.label+"' type='text' name='"+this.name+"' id='"+this.control_id()+"' value=\""+this.properties.value+"\" /></div>";
	this.jquery_place().replaceWith(html);

	var self = this;
	this.jquery().change(function(){
		self.basewidget.fire_event( this.name, 'changed' );
	});

	if(this.properties.visible) 
		this.jquery_place().show();
	else 
		this.jquery_place().hide();
};

////////////////////////////////////////////////////////////////////////////////
